optuna/study/_multi_objective.py

```diff
@@ -91,26 +91,31 @@ def _fast_non_dominated_sort(
                 len(penalty), len(objective_values)
             )
         )
-    nondomination_rank = np.zeros(len(objective_values), dtype=int)
+    nondomination_rank = np.full(len(objective_values), -1)
```

Comment:
As we can already bound `max(nondomination_rank)` by `n_below` and `nondomination_rank` of `n_below + 1` will not be used, so what about using `n_below + 1`?
Another reason why we should probably avoid `-1` is that it might cause unexpected bugs in the future when some developers use `nondomination_rank` being always better when it is lower.
Plus, this implementation requires an ad-hoc handling of `nondomination_rank=-1` in each place where the function is used.